### ISSUE TITLE
Handling of partial reliable protected user messages

### ISSUE THREAD

**Issue by gloinul** [gloinul/draft-westerlund-tsvwg-dtls-over-sctp-bis#28]
So the partial delivery of user messages will create situation where some DTLS records in a sequence of a protected used message will not be delivered in its entirety. Thus, I think we need to add some words on how one handles this situation. 

First, there is the question of ensuring that any buffer containing data that contains partial DTLS record are flushed. 
Secondly, if one have a partial delivery API towards the application, then that needs to get a error indication that this user message was not delivered in its entirety. 

These issues are much more significant for this spec than it was for RFC 6083, where the DTLS record needed to be complete to decrypt and pass integrity. In this case one or more DTLS records from the start might be delivered.

**Comment by tuexen** [comment]
Question: Why do we need to deal with the upper layer API of DTLS at all? Normally the IETF doesn't do it at all and, I think, doing the socket API for SCTP was a special thing.

**Comment by tuexen** [comment]
Doesn't imply
```
DTLS/SCTP, automatically fragments and reassembles user messages
```
atomic `send()`/`recv()` calls?

**Comment by gloinul** [comment]
I am concerned with the SCTP<-> DTLS interactions due to partially receiving protected user message. The DTLS layer with the current definition is dependent on the in sequence delivery of the DTLS records and also the indication that the complete protected user message. 

In cases where not all DTLS records of a protected user messages is delivered it is vital that the DTLS layer knows that and either delivers nothing (atomic operations) or indicates that only partial delivery have occurred if some parts of the user message has been decrypted and delivered.

**Comment by tuexen** [comment]
Sure, I agree.

**Comment by teiclap** [comment]
Text has been adapted for covering both the case where partial delivery is supported and where it isn't.

**Comment by gloinul** [comment]
So this is being resolved by #40.

**Comment by gloinul** [comment]
To my understanding this can be closed. In #48 a paragraph was added to also ensure that PR-SCTP don't result in terminated SCTP association when failures.
